tests/phpunit/tests/query/searchColumns.php

```diff
@@ -0,0 +1,381 @@
+<?php
+/**
+ * Testing the search columns support in `WP_Query`.
+ *
+ * @package WordPress\UnitTests
+ * @since 6.2.0
+ */
+
+/**
+ * Test cases for the search columns feature.
+ *
+ * @group query
+ * @group search
+ *
+ * @since 6.2.0
+ */
+class Tests_Query_SearchColumns extends WP_UnitTestCase {
+	/**
+	 * The post ID of the first fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid1
+	 */
+	protected static $pid1;
+
+	/**
+	 * The post ID of the second fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid2
+	 */
+	protected static $pid2;
+
+	/**
+	 * The post ID of the third fixture post.
+	 *
+	 * @since 6.2.0
+	 * @var int $pid3
+	 */
+	protected static $pid3;
+
+	/**
+	 * Create posts fixtures.
+	 *
+	 * @param WP_UnitTest_Factory $factory The factory instance.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$pid1 = self::factory()->post->create(
+			array(
+				'post_status'  => 'publish',
+				'post_title'   => 'foo title',
+				'post_excerpt' => 'foo excerpt',
+				'post_content' => 'foo content',
+			)
+		);
+		self::$pid2 = self::factory()->post->create(
+			array(
+				'post_status'  => 'publish',
+				'post_title'   => 'bar title',
+				'post_excerpt' => 'foo bar excerpt',
+				'post_content' => 'foo bar content',
+			)
+		);
+
+		self::$pid3 = self::factory()->post->create(
+			array(
+				'post_status'  => 'publish',
+				'post_title'   => 'baz title',
+				'post_excerpt' => 'baz bar excerpt',
+				'post_content' => 'baz bar foo content',
+			)
+		);
+	}
+
+	/**
+	 * The search should use default search columns when search columns are empty.
+	 */
+	public function test_s_should_use_default_search_columns_when_empty_search_columns() {
+		$q = new WP_Query(
+			array(
+				's'              => 'foo',
+				'search_columns' => array(),
+				'fields'         => 'ids',
+			)
+		);
+
+		$this->assertStringContainsString( 'post_title', $q->request, 'SQL request should contain post_title string.' );
+		$this->assertStringContainsString( 'post_excerpt', $q->request, 'SQL request should contain post_excerpt string.' );
+		$this->assertStringContainsString( 'post_content', $q->request, 'SQL request should contain post_content string.' );
+		$this->assertEqualSets( array( self::$pid1, self::$pid2, self::$pid3 ), $q->posts, 'Query results should be equal to the set.' );
```

Comment:
```suggestion
		$this->assertSameSetsWithIndex( array( self::$pid1, self::$pid2, self::$pid3 ), $q->posts, 'Query results should be equal to the set.' );
```
As `self::$pid1`, `self::$pid2`, and `self::$pid3` are `int`, this assertion can be stricter.

🔢 Applies to all instances in the PR.

Reply:
Assertions updated.